libr/core/cmd.c

```diff
@@ -1191,8 +1191,43 @@ static int cmd_system(void *data, const char *input) {
 	return ret;
 }
 
+static void
+prepare_intgrep(char *cmd, const char *quotestr)
+{
+	char *ptr;
+	int len = strlen (cmd);
+
+	//  search and modify '~str~?' into '~?str'
+	if (r_str_endswith (cmd, "~?") && len > 4 && cmd[len - 3] != '\\') {
```

Comment:
check length first because that doesnt implies any string processing so you can skip false positives asap